src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -805,6 +809,11 @@ private function addPriceInformation(ProductPresentationSettings $settings, arra
      */
     protected function shouldEnableAddToCartButton(array $product, ProductPresentationSettings $settings)
     {
+        // If the product is discontinued and is displayed, we disable add to cart button
+        if ($product['active'] != 1) {
```

Comment:
the code doesn't seem to have any relation with the comment?
"discontinued and displayed" is mentioned, but the code only checks if product is active :thinking:

Reply:
What I meant was - if the product is inactive and you chose the redirect option "keep displayed"

Reply:
yeah but by looking at the code there is nothing related to the redirection, it only disables the button if product is inactive, so I think the comment should be removed or changed to smth like `if the product is disabled, we disable the addToCartButton too` (but I think its self explanatory, so you can just get rid of the comment)

Reply:
OK! Good idea. :-)